assets/src/edit-story/components/panels/document/publish/publishTime.js

```diff
@@ -77,6 +79,13 @@ function PublishTime() {
     [showDatePicker, updateStory]
   );
 
+  // Floating date means an unset date so that the story publish date will match the time it will get published.
+  const floatingDate =
+    ['draft', 'pending', 'auto-draft'].includes(status) &&
+    (date === modified || date === null);
+  const displayLabel = !floatingDate
+    ? format(date, shortDateFormat) + ' ' + formatTime(date)
+    : __('Immediately', 'web-stories');
```

Comment:
For some reason I thought Gutenberg does this differently but apparently it doesn't 🤷 

https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/editor/src/store/selectors.js#L745-L775

TIL